spec/models/vacation_request_spec.rb

```diff
@@ -0,0 +1,94 @@
+require 'rails_helper'
+
+RSpec.describe VacationRequest do
+  it 'has a valid factory' do
+    vacation_request = FactoryGirl.build(:vacation_request)
+
+    expect(vacation_request).to be_valid
+  end
+
+  context 'as a brand new object' do
+    let(:vacation_request) { VacationRequest.new }
+
+    it { expect(vacation_request).to have_attributes duration: nil }
+    it { expect(vacation_request).to have_attributes end: nil }
+    it { expect(vacation_request).to have_attributes kind: 'planned' }
+    it { expect(vacation_request).to have_attributes status: 'requested' }
+    it { expect(vacation_request).to have_attributes start: nil }
+    it { expect(vacation_request).to have_attributes user_id: nil }
+
+    it { expect(vacation_request).not_to be_valid }
+  end
+
+  it 'allows to pass date in ISO 8601 format, that is, YYYY-MM-DD' do
+    vacation_request = FactoryGirl.build(:vacation_request, start: '2017-05-01')
+
+    expect(vacation_request).to be_valid
+  end
+
+  it 'allows to pass date as a Ruby Date object' do
+    vacation_request = FactoryGirl.build(:vacation_request,
+                                         start: Time.zone.today)
+
+    expect(vacation_request).to be_valid
+  end
+
+  it 'does not allow to pass "duration" equal to 0' do
+    vacation_request = FactoryGirl.build(:vacation_request, duration: 0)
+
+    expect(vacation_request).not_to be_valid
+  end
+
+  it 'does not allow to pass "duration" as a float, like 0.5' do
+    vacation_request = FactoryGirl.build(:vacation_request, duration: 5.5)
+
+    expect(vacation_request).not_to be_valid
+  end
+
+  it 'does not allow to pass "end" as incorrect date when "status=used"' do
+    vacation_request = FactoryGirl.build(:vacation_request, status: 'used')
+
+    expect(vacation_request).not_to be_valid
+  end
+
+  describe '.requested_accepted_inprogress' do
+    let(:user) { FactoryGirl.create :user, :with_vacations_of_all_statuses }
+
+    it 'provides accordingly filtered list of vacation requests' do
+      expect(user.vacation_requests.count).to be(6)
```

Comment:
We usually use .to eq(..) for strings and numbers (http://stackoverflow.com/questions/7156955/whats-the-difference-between-equal-eql-and)